src/LineEnding.elm

```diff
@@ -0,0 +1,62 @@
+module LineEnding exposing (optionalWhitespaceUpToOneLineEnding)
+
+import Helpers
+import Parser
+import Parser.Advanced as Advanced exposing (..)
+
+
+type alias Parser a =
+    Advanced.Parser String Parser.Problem a
+
+
+optionalWhitespaceUpToOneLineEnding : Parser ()
+optionalWhitespaceUpToOneLineEnding =
+    loop CanHaveLineEnding statementsHelp
+
+
+type State
+    = CanHaveLineEnding
+    | CannotHaveLineEnding
+
+
+statementsHelp : State -> Parser (Step State ())
+statementsHelp state =
+    oneOf
+        [ lineEnding
+            |> andThen
+                (\() ->
+                    case state of
+                        CanHaveLineEnding ->
+                            Loop CannotHaveLineEnding |> succeed
+
+                        CannotHaveLineEnding ->
+                            problem (Parser.Problem "Can't have more than one line ending character.")
+                )
+        , succeed (Loop state)
+            |. Advanced.chompIf Helpers.isGfmWhitespace (Parser.Expecting "Whitespace")
+        , succeed (Done ())
+        ]
+
+
+{-| From <https://spec.commonmark.org/0.29/#line-ending>:
+
+> is a newline (U+000A), a carriage return (U+000D) not followed by a newline, or a carriage return and a following newline.
+
+-}
+lineEnding : Parser ()
```

Comment:
Should this section just reference the definition `src/Helpers`? I apologize if there's a difference that I missed - it looks like a copy to me.

Reply:
They could be combined as some kind of `ControlCharacters` file